packages/core/saved-objects/core-saved-objects-server-internal/src/status.ts

```diff
@@ -80,3 +44,43 @@ export const calculateStatus$ = (
     })
   );
 };
+
+const migratorStatusToServiceStatus = (
+  migrationStatus: KibanaMigratorStatus
+): ServiceStatus<SavedObjectStatusMeta> => {
+  if (migrationStatus.status === 'waiting_to_start') {
+    return {
+      level: ServiceStatusLevels.unavailable,
+      summary: `SavedObjects service is waiting to start migrations`,
+    };
+  } else if (migrationStatus.status === 'waiting_for_other_nodes') {
+    return {
+      level: ServiceStatusLevels.unavailable,
+      summary: `SavedObjects service is waiting for other nodes to complete the migration`,
+      detail:
+        `If no other Kibana instance is attempting ` +
+        `migrations, you can get past this message by deleting index ${migrationStatus.waitingIndex} and ` +
+        `restarting Kibana.`,
+    };
+  } else if (migrationStatus.status === 'running') {
+    return {
+      level: ServiceStatusLevels.unavailable,
+      summary: `SavedObjects service is running migrations`,
+    };
+  }
+
+  const statusCounts: SavedObjectStatusMeta['migratedIndices'] = { migrated: 0, skipped: 0 };
+  if (migrationStatus.result) {
+    migrationStatus.result.forEach(({ status }) => {
+      statusCounts[status] = (statusCounts[status] ?? 0) + 1;
+    });
+  }
+
+  return {
+    level: ServiceStatusLevels.available,
+    summary: `SavedObjects service has completed migrations and is available`,
```

Comment:
I think we should be ok, but let's see what @rudolf thinks about it. I'm fine changing the impl to use another/new status instead (as of https://github.com/elastic/kibana/pull/145693#discussion_r1026411879)

Reply:
This is a undocumented, worst-case scenario workaround, not really a feature users should rely on long term. So I wouldn't spend a lot of time on polishing the status message copy, but making it explicit that migrations were skipped feels like the more ideal end-state. So if it's quick let's add it, otherwise happy to merge as-is :+1: